src/cmd/manage_nsfs.js

```diff
@@ -592,24 +604,35 @@ async function list_config_files(type, wide, show_secrets, filters = {}) {
     // in case we have a filter by name, we don't need to read all the entries and iterate them
     // instead we "mock" the entries array to have one entry and it is the name by the filter (we add it for performance)
     if (is_filter_by_name) {
-        entries = [filters.name];
+        entry_names = [filters.name];
     } else if (type === TYPES.ACCOUNT) {
-        entries = await config_fs.list_accounts();
+        entry_names = await config_fs.list_accounts();
     } else if (type === TYPES.BUCKET) {
-        entries = await config_fs.list_buckets();
+        entry_names = await config_fs.list_buckets();
     }
 
-    let config_files_list = await P.map_with_concurrency(10, entries, async entry => {
+    // temporary cache for mapping bucker owner_account (id) -> bucket_owner (name)
+    const bucket_owners_map = {};
+    let config_files_list = await P.map_with_concurrency(10, entry_names, async entry_name => {
         if (wide || should_filter) {
             const data = type === TYPES.ACCOUNT ?
-                await config_fs.get_account_by_name(entry, options) :
-                await config_fs.get_bucket_by_name(entry, options);
+                await config_fs.get_account_by_name(entry_name, options) :
+                await config_fs.get_bucket_by_name(entry_name, options);
             if (!data) return undefined;
             if (should_filter && !filter_list_item(type, data, filters)) return undefined;
             // remove secrets on !show_secrets && should filter
-            return wide ? _.omit(data, show_secrets ? [] : ['access_keys']) : { name: entry };
+            if (!wide) return { name: entry_name };
+            if (type === TYPES.ACCOUNT) return _.omit(data, show_secrets ? [] : ['access_keys']);
+            if (type === TYPES.BUCKET) {
+                data.bucket_owner = bucket_owners_map[data.owner_account];
+                if (!data.bucket_owner) {
+                    await set_bucker_owner(data);
+                    bucket_owners_map[data.owner_account] = data.bucket_owner;
+                }
```

Comment:
I think that reusing the `bucket_owner` as it was saved in the bucket config will not be valid in case the account was renamed, so I'm not sure that you can really reduce the system call (to read the account config).

Reply:
we can say it also when inside the loop, we checked the first account but the account name was changed or new accounts were added... I think it's fine since it's list

Reply:
You are right, but a property of `bucket_owner` in a bucket might be not updated for years/months, unlike a bucket that was created during the loop which is not updated for seconds.

For example, the account and bucket were created in version 5.15 and in version 5.16 the account was renamed (bucket config was not updated), in version 5.18 when listing the buckets I think we should expect to see the bucket owner's updated name (and not the one which was originally saved).

Reply:
I think there is a missunderstnding, i'm not using the data.bucket_owner old property, inside set_bucket_owner(data) I read the account config by id and use it

Reply:
Now I saw the map and the comment, thanks.